src/components/utils/popover/popover-mobile.ts

```diff
@@ -0,0 +1,142 @@
+import { PopoverAbstract } from './popover-abstract';
+import ScrollLocker from '../scroll-locker';
+import { PopoverHeader } from './components/popover-header';
+import { PopoverStatesHistory } from './utils/popover-states-history';
+import { PopoverParams } from './popover.types';
+import { PopoverItem } from './components/popover-item';
+import { PopoverItem as PopoverItemParams } from '../../../../types';
+import { css } from './popover.const';
+import Dom from '../../dom';
+
+/**
+ * Mobile Popover.
+ * On mobile devices Popover behaves like a fixed panel at the bottom of screen. Nested item appears like "pages" with the "back" button
+ */
+export class PopoverMobile extends PopoverAbstract {
+  /**
+   * ScrollLocker instance
+   */
+  private scrollLocker = new ScrollLocker();
+
+  /**
+   * Reference to popover header if exists
+   */
+  private header: PopoverHeader | undefined | null;
+
+  /**
+   * History of popover states for back navigation.
+   * Is used for mobile version of popover,
+   * where we can not display nested popover of the screen and
+   * have to render nested items in the same popover switching to new state
+   */
+  private history = new PopoverStatesHistory();
+
+  /**
+   * Construct the instance
+   *
+   * @param params - popover params
+   */
+  constructor(params: PopoverParams) {
+    super(params);
+
+    this.nodes.overlay = Dom.make('div', [css.overlay, css.overlayHidden]);
+    this.nodes.popover.insertBefore(this.nodes.overlay, this.nodes.popover.firstChild);
+
+    this.listeners.on(this.nodes.overlay, 'click', () => {
+      this.hide();
+    });
+
+    /* Save state to history for proper navigation between nested and parent popovers */
+    this.history.push({ items: params.items });
+  }
+
+  /**
+   * Open popover
+   */
+  public show(): void {
+    this.nodes.overlay?.classList.remove(css.overlayHidden);
```

Comment:
```suggestion
    this.nodes.overlay!.classList.remove(css.overlayHidden);
```

or pass `PopoverNodes` as generic to `PopoverAbstract` to explicitly tell ts, that `PopoverMobile` has overlay